linetools/isgm/emsystem.py

```diff
@@ -0,0 +1,536 @@
+          Emission redshift
+        zem : float
+          Emission redshift of background source
+        vlim : Quantity array (2)
+          Velocity limits of the system
+        ZH :  float
+          Metallicity (log10)
+        name : str
+            Name of the system
+    """
+
+    __metaclass__ = ABCMeta
+
+    @classmethod
+    def from_emlines(cls, emlines, vlim=None, **kwargs):
+        """Instantiate from a list of EmLines
+
+        Parameters
+        ----------
+        emlines : list
+          List of EmLine objects
+        vlim : list, optional
+          Velocity limits for the system
+          If not set, the first components sets vlim
+        """
+        # Check
+        #assert ltiu.chk_components(components)
+        # Instantiate with the first component
+        init_eml = emlines[0]
+        #
+        slf = cls(init_eml.attrib['coord'], init_eml.z, vlim=vlim)
+        if slf.chk_emline(init_eml):
+            slf._emlines.append(init_eml)
+        else:
+            raise IOError("Bad component input")
+        # Append with component checking
+        for emline in emlines[1:]:
+            slf.add_emline(emline)
+        # Return
+        return slf
+
+    @classmethod
+    def from_alis(cls, alis_file, radec, **kwargs):
+        """ Load from an ALIS output file
+
+        Parameters
+        ----------
+        alis_file
+        radec : SkyCoord or tuple of RA,DEC
+        kwargs
+
+        Returns
+        -------
+        EmSystem
+
+        """
+        emlines = lio.emlines_from_alis_output(alis_file)
+        # Add coordinates
+        coord = ltu.radec_to_coord(radec)
+        for emline in emlines:
+            emline.attrib['coord'] = coord
+        #
+        return cls.from_emlines(emlines, **kwargs)
+
+    @classmethod
+    def from_dict(cls, idict, use_coord=False, **kwargs):
+        """ Instantiate from a dict.  Usually read from the hard-drive
+
+        Parameters
+        ----------
+        idict : dict
+
+        Returns
+        -------
+
+        """
+        slf = cls(SkyCoord(ra=idict['RA']*u.deg, dec=idict['DEC']*u.deg),
+                  idict['zem'], vlim=idict['vlim']*u.km/u.s,
+                  name=idict['Name'], em_type=idict['em_type'])
+        # Emission lines
+        em_dict = idict['emlines']
+        for key in em_dict:
+            iline = em_dict[key]
+            obj = EmLine.from_dict(iline)
+            slf.add_emline(obj)
+        # Return
+        return slf
+
+    @classmethod
+    def from_json(cls, json_file, **kwargs):
+        """ Load from a JSON file (via from_dict)
+
+        Parameters
+        ----------
+        json_file
+        kwargs
+
+        Returns
+        -------
+        AbsSystem
+
+        """
+        idict = ltu.loadjson(json_file)
+        slf = cls.from_dict(idict, **kwargs)
+        return slf
+
+
+    def __init__(self, radec, zem, vlim=None, em_type=None, name=None):
+
+        self.zem = zem
+        if vlim is None:
```

Comment:
We need to use the zLimits Class here.

I can help set that up.  Remind me.

Reply:
This is now in Issue #419